pkg/rehearse/jobs.go

```diff
@@ -219,9 +230,82 @@ func ConfigureRehearsalJobs(toBeRehearsed config.Presubmits, ciopConfigs config.
 		}
 	}
 
+	// for the remaining cluster types of each template, pick a `random` corresponding job to rehearse.
+	for templateFile, clusterTypes := range templatesWithClusterTypes {
+		for clusterType := range clusterTypes {
+			if job := pickTemplateJob(prConfig.JobConfig.Presubmits, templateFile, clusterType); job != nil {
+
+				// change the configmap name with the temporary created one.
+				exists, index, _ := hasChangedTemplateVolume(job.Spec.Containers[0].VolumeMounts, job.Spec.Volumes, templates)
+				if exists {
+					templateData, err := config.GetTemplateData(templates[templateFile])
+					if err != nil {
+						loggers.Job.WithError(err).WithField("template-name", templates[templateFile].Name).Warn("couldn't get template's data. Job won't be rehearsed")
+						continue
+					}
+					job.Spec.Volumes[index].VolumeSource.ConfigMap.Name = config.GetTempCMName(config.GetTemplateName(templateFile), templateFile, templateData)
+				}
+
+				rehearsals = append(rehearsals, job)
+			}
+		}
+	}
+
 	return rehearsals
 }
 
+func getTemplatesWithClusterTypes(templates config.CiTemplates) map[string]sets.String {
+	templatesKeys := make(map[string]sets.String)
+	for filename := range templates {
+		templatesKeys[filename] = sets.NewString([]string{"aws", "gcs", "openstack", "libvirt"}...)
+	}
+	return templatesKeys
+}
+
+func pickTemplateJob(presubmits map[string][]prowconfig.Presubmit, templateFile, clusterType string) *prowconfig.Presubmit {
+	for _, jobs := range presubmits {
+		for _, job := range jobs {
+			if job.Agent != string(pjapi.KubernetesAgent) {
+				continue
+			}
+
+			if hasClusterType(job, clusterType) {
+				if ok := hasTemplateFile(job, templateFile); ok {
+					for index, env := range job.Spec.Containers[0].Env {
+						if env.Name == testCommandEnvName {
```

Comment:
So what happens if we don't have a TEST_COMMAND env?

Reply:
removed.